redash/utils/parameterized_query.py

```diff
@@ -1,9 +1,29 @@
 import pystache
+from flask_login import current_user
 from numbers import Number
-from redash.utils import mustache_render
+from redash import models
+from redash.utils import mustache_render, json_loads
+from redash.permissions import require_access, view_only
 from funcy import distinct
 from dateutil.parser import parse
 
+
+def dropdown_values(query_id, org):
+    def _pluck_name_and_value(row):
+        name_column = "name" if "name" in row.keys() else row.keys()[-1]
```

Comment:
In the JS version we were:

1. Performing case insensitive search (converting row keys to lower case).
2. Using the _first_ column name by default (-1 is the last, no?).

Reply:
1. Roger that.
2. It is taking the first column (see [this test](https://github.com/getredash/redash/pull/3337/files#diff-373dcffefcbd523d468cf8afa3a45720R176)). I believe this has to do with the order of key additions in `json_loads`. Anyway, it works as expected.

Reply:
> It is taking the first column (see this test). I believe this has to do with the order of key additions in json_loads. Anyway, it works as expected.

It's not. Try changing the JSON into:

```
'{"rows": [{"zid": 5, "fish": "Clown", "poultry": "Hen"}]}'
```

You will notice that `zid` is no longer the last entry, but the 2nd (⁉️).

The order of keys in a Python dictionary can't be trusted. Considering that in the view we order the columns by their order in the `columns` array, maybe we should just use that here as well?